### ISSUE TITLE
Age based mortality

### ISSUE THREAD

**Issue by krosenfeld-IDM** [InstituteforDiseaseModeling/laser#28]
On commit [9a1ba81](https://github.com/InstituteforDiseaseModeling/laser/commit/9a1ba81ae2371cc7f7d71f9716a416c239941621) we're seeing non disease mortality spike at the beginning of the simulation:
![image](https://github.com/user-attachments/assets/c14c188e-edfd-4924-a165-55ce2396034f)

This may be due to `dod`  not being offest by current `tick` as agents are activated:
```
model.population.dod[istart:iend] = pdsod(model.population.dob[istart:iend], max_year=100)   # make use of the fact that dob[istart:iend] is currently 0 
```
see https://github.com/InstituteforDiseaseModeling/laser/blob/clorton/cleanup-for-merge/nnmm/measles.ipynb

**Comment by clorton** [comment]
I am still seeing this behavior even with the fix in `do_births()`.

**Comment by krosenfeld-IDM** [comment]
That's a good point @KevinMcCarthyAtIDM. I think we're using US statistics for this notebook https://github.com/InstituteforDiseaseModeling/laser/blob/e6b00057ead082c8f38e3daf2fa0c6f97d4bb58a/src/idmlaser/kmcurve.py#L6

so the effect should be even larger if we switch to NGA statistics. I have what should be the equivalent table in the GEOMED repository if you want to try that @clorton

**Comment by krosenfeld-IDM** [comment]
I'm seeing the same effect in the current GEOMED version (https://github.com/gatesfoundation/GEOMED24/commit/0dea1e11d26ded21eda466da37e82b17cc67e72f) although the yearly deaths seem to rise linearly:

![image](https://github.com/user-attachments/assets/bd3a4904-3b4e-4391-b37a-b90ca6f2f8ff)

https://github.com/gatesfoundation/GEOMED24/commit/0dea1e11d26ded21eda466da37e82b17cc67e72f

I think it's likely that the first year will be part of a burn-in anyway and the effect goes away after that. But good to know about and glad it is documented here! Nice catch @KevinMcCarthyAtIDM .

**Comment by KevinMcCarthyAtIDM** [comment]
> I would expect yearly deaths to increase as the population increases and is not offset by any increasing average lifespan.

Agreed, that's normal.  But the spike in year 1 isn't.  This is probably not a breaking bug for Katherine's work, since it disappears after burn-in.  But also of course I'd feel better making sure we understand it.  I think the quickest path is looking at the age distributions of all agents who die in Year 1 vs. all agents who die in some later year.  The explanations I gave above would imply that the histogram in Year 1 should have more kids in the 1-2 year age bin than later years.  If we see that, great, we understand it enough to know it's not the most critical fix for now; but if we don't see that, then I'm confused and want to make sure it's not something weirder...

**Comment by krosenfeld-IDM** [comment]
Here is what I'm seeing for the age distributions @KevinMcCarthyAtIDM comparing the first year of the sim and the rest:

![image](https://github.com/user-attachments/assets/e04e80d1-1fa5-436c-8d34-3af5bb6fac3a)


https://github.com/gatesfoundation/GEOMED24/blob/features/notebooks/measles.py

**Comment by krosenfeld-IDM** [comment]
ok - here are the deaths as measured by:

https://github.com/gatesfoundation/GEOMED24/blob/2cadae686ee80f1967eb5e7a620dd1b00a939540/notebooks/measles.py#L633

![image](https://github.com/user-attachments/assets/0503d3b4-f1f4-4114-93f0-de543bc6f353)

Distributions look fairly similar.

**Comment by krosenfeld-IDM** [comment]
Updating with NGA deaths and age pyramid (and plotting density for readability)
![image](https://github.com/user-attachments/assets/ca7aa7b6-23fe-48a3-8abd-9b33cd3a6de5)
https://github.com/gatesfoundation/GEOMED24/blob/922c64731e6bc3de4cfb1b6767479da9a4ca949a/notebooks/measles.py
